### PR TITLE
Added lombok, created entities, created basic controller for users

### LINKED ISSUE #7: Add REST API for CRUD operations
---
### Description
Add a controller with basic CRUD operations: read for all users and the current one
All handler methods can be added without particular implementations so far. Just leave `TODO`

### Links to read:

- [DispatcherServlet](https://docs.spring.io/spring-framework/docs/current/reference/html/web.html#mvc-servlet);
- [Context Hierarchy](https://docs.spring.io/spring-framework/docs/current/reference/html/web.html#mvc-servlet-context-hierarchy);
- [Annotated controllers](https://docs.spring.io/spring-framework/docs/current/reference/html/web.html#mvc-controller) (only Declaration, Request Mapping, and Handler methods)

### Definition of Done (DoD):

- [x] Code reviewed
- [x] PR merged
- [x] Sending request to each API method (via [Postman](https://www.postman.com/)) succeed

### PR DESCRIPTION
**Pull request by COCONUTmkII**
Added Lombok, created entities created a basic controller for users, and created basic security configuration.

### CODE REVIEW

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ spring:
     hibernate:
       ddl-auto: none
   datasource:
-    url: jdbc:postgresql://localhost:8081/userapi
+    url: jdbc:postgresql://localhost:8081/userapi_app
```
Comment (Dauren-Delmukhambetov) [inline src/main/resources/application.yaml:8]: Why did you change the database name?

File: src/main/java/by/itechart/api/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/user")
```
Comment (Dauren-Delmukhambetov) [inline src/main/java/by/itechart/api/controller/UserController.java]: Why did you decide to use a singular form of  `user`?

File: src/main/java/by/itechart/api/entity/User.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import lombok.Getter;
+import lombok.Setter;
+import javax.persistence.*;
+import java.time.LocalDate;
+
+@Entity
+@Table(name = "users")
+@Getter
+@Setter
```
Comment (Dauren-Delmukhambetov) [inline src/main/java/by/itechart/api/entity/User.java]: Not calling to change but just sharing the information: the `@Data` [annotation](https://projectlombok.org/features/Data) of Lombok includes both `@Getter` and `@Setter`

File: src/main/java/by/itechart/api/security/SecurityConfig.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package by.itechart.api.security;
```
Comment (Dauren-Delmukhambetov) [inline src/main/java/by/itechart/api/security/SecurityConfig.java]: Overall, package naming is good. I like it! 
In this particular case it seems logical to me but moving further I think we will have more configuration classes and it would be better to keep them all in the `configuration` package. WDYT?

**Comment by Dauren-Delmukhambetov** [comment]
Description in the task sounds a bit confusing, sorry for that. But in the scope of this task Create, Update, and Delete methods should be covered too.
You can cover them in the next PR if you like

File: src/main/java/by/itechart/api/controller/UserController.java (hunk 2 above)
Comment (COCONUTmkII) [reply src/main/java/by/itechart/api/controller/UserController.java]: I think that every operation here will be related to a user, that's why I decided to do that.

File: src/main/resources/application.yaml (hunk 1 above)
Comment (COCONUTmkII) [reply src/main/resources/application.yaml:8]: There was a problem with the connection with it. The database name was incorrect, that's why I changed it.

File: src/main/java/by/itechart/api/entity/User.java (hunk 3 above)
Comment (COCONUTmkII) [reply src/main/java/by/itechart/api/entity/User.java]: Thank's!

File: src/main/java/by/itechart/api/security/SecurityConfig.java (hunk 4 above)
Comment (COCONUTmkII) [reply src/main/java/by/itechart/api/security/SecurityConfig.java]: Seems legit! I will change the package level to `configuration -> security`.

**Review by Dauren-Delmukhambetov — APPROVED** [review]
Well done!
